### ISSUE TITLE
Alpha.6 Server crash on picking up item from stockpile

### ISSUE THREAD

**Issue by Fank** [StrangeLoopGames/EcoIssues#2393]
![image](https://user-images.githubusercontent.com/1900106/31012868-c3a32a68-a513-11e7-95aa-17c1249d3239.png)

```
Unhandled Exception: System.AggregateException: One or more errors occurred. ---> System.InvalidOperationException: Changes could not be applied, the requested operation would result in an invalid inventory state.
   at Eco.Gameplay.Items.InventoryChangeSet.ApplyInternal(Boolean throwOnInvalid)
   at Eco.Gameplay.Items.Inventory.Modify(Action`1 callback, User user)
   at Eco.Gameplay.Components.StockpileComponent.Tick(Single dtime)
   at System.Collections.Generic.List`1.ForEach(Action`1 action)
   at Eco.Gameplay.Objects.WorldObject.Tick(Single dtime)
   at System.Threading.Tasks.Parallel.<>c__DisplayClass42_0`2.<PartitionerForEachWorker>b__1()
   at System.Threading.Tasks.Task.InnerInvokeWithArg(Task childTask)
   at System.Threading.Tasks.Task.<>c__DisplayClass176_0.<ExecuteSelfReplicating>b__0(Object )
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Parallel.PartitionerForEachWorker[TSource,TLocal](Partitioner`1 source, ParallelOptions parallelOptions, Action`1 simpleBody, Action`2 bodyWithState, Action`3 bodyWithStateAndIndex, Func`4 bodyWithStateAndLocal, Func`5 bodyWithEverything, Func`1 localInit, Action`1 localFinally)
   at System.Threading.Tasks.Parallel.ForEachWorker[TSource,TLocal](IEnumerable`1 source, ParallelOptions parallelOptions, Action`1 body, Action`2 bodyWithState, Action`3 bodyWithStateAndIndex, Func`4 bodyWithStateAndLocal, Func`5 bodyWithEverything, Func`1 localInit, Action`1 localFinally)
   at System.Threading.Tasks.Parallel.ForEach[TSource](IEnumerable`1 source, Action`1 body)
   at Eco.Gameplay.Objects.WorldObjectManager.Run()
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.ThreadHelper.ThreadStart(Object obj)
```

**Comment by MarjieAVolk** [comment]
Actually, did you already fix this?

**Comment by Fank** [comment]
@MarjieAVolk addressing @metachronism ?

**Comment by metachronism** [comment]
fixed in 6.0.1
